### DISCUSSION TITLE
404 errors when going via Nginx Ingress

### DISCUSSION THREAD

**Discussion by devopstom-user** [keycloak/keycloak#10542] [Archive]
Hi,

deployed keycloak v17.0.0 to my minikube and it works fine when I use port-forward directly to the SVC IP, like this:
`kubectl port-forward svc/keycloak-svc 8080:8080 &`
However going via Nginx Ingress results in a number of 404 errors for some of the page resources(seen in browser dev tools net panel), but some get 200 OK. This results  page doesn't load correctly and looks completely broken. Probably missing something somewhere in the cfg, would appreciate any hints on this!

I start the container with the following options:
```      containers:
      - args:
        - start-dev
        - --hostname=keycloak.10.96.38.235.nip.io
        - --http-enabled=true
        - --http-relative-path="/auth"
        - --proxy=edge
```
I tried various options for proxy from the below link, but I always see same problem with many 404
https://www.keycloak.org/server/reverseproxy

The Ingress resource looks like this:
``` apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    ingress.kubernetes.io/proxy-body-size: 128M
    kubernetes.io/ingress.class: nginx
  name: test-keycloak-ingress
  namespace: test-keycloak
spec:
  rules:
  - host: keycloak.10.96.38.235.nip.io
    http:
      paths:
      - backend:
          service:
            name: keycloak-svc
            port:
              number: 8080
        path: /
        pathType: Prefix
```
Many thanks!

**Comment by devopstom-user** [comment]
surprisingly it starts to work fine, when I change replica count from 2 to 1 for the sts:
`kubectl -n test-keycloak scale statefulset test-keycloak --replicas=1`
so it seem this has to do with session affinity but nothing is mentioned in the guide wrt using reverse proxy.
I tried with these annotations but it did not help:
```
    nginx.ingress.kubernetes.io/affinity: cookie
    nginx.ingress.kubernetes.io/session-cookie-name: AUTH_SESSION_ID_LEGACY
```
any suggestions how this should be setup for clustered ha like setup with multiple pods?

**Reply by fabiogoma** [reply]
Have you tried this:
```yaml
JAVA_OPTS_APPEND: "-Djgroups.dns.query=keycloak.default.svc.cluster.local"
```
https://www.keycloak.org/server/caching#_available_transport_stacks
